lib/src/picture_provider.dart

```diff
@@ -578,7 +585,8 @@ class NetworkPicture extends PictureProvider<NetworkPicture> {
     }
     return other is NetworkPicture &&
         url == other.url &&
-        colorFilter == other.colorFilter;
+        colorFilter == other.colorFilter &&
+        theme == other.theme;
   }
 
   @override
```

Comment:
OVerriding hashcode is no longer valid. Argh. If we have a mutable property on here it's going to break the caching logic, which wants to use this as a key.